net/dns/manager_darwin.go

```diff
@@ -36,15 +36,11 @@ func (c *darwinConfigurator) SupportsSplitDNS() bool {
 func (c *darwinConfigurator) SetDNS(cfg OSConfig) error {
 	var buf bytes.Buffer
 	buf.WriteString(macResolverFileHeader)
-	for i, ip := range cfg.Nameservers {
-		if i == 0 {
-			buf.WriteString("nameserver ")
-		} else {
-			buf.WriteString(" ")
-		}
+	for _, ip := range cfg.Nameservers {
+		buf.WriteString("nameserver ")
```

Comment:
Ideally setting nameservers should all be done via SystemConfiguration so the conf is properly saved in the macOS Dynamic Store (ie: do this via programatic equivalent of scutil or spawn scutil itself with an input file).  Writing  directly to /etc/resolver does work as a fallback though.

Reply:
Verfied that the correct format is:
nameserver 1.1.1.1
nameserver 8.8.8.8

(new line for each IP)

% scutil --dns
...
resolver #13
  domain   : ts
  nameserver[0] : 1.1.1.1
  nameserver[1] : 8.8.8.8